sdk/textanalytics/Azure.AI.TextAnalytics/src/DynamicClassifyOptions.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+
+namespace Azure.AI.TextAnalytics
+{
+    /// <summary>
+    /// A set of options used to configure dynamic classification.
+    /// </summary>
+    public class DynamicClassifyOptions : TextAnalyticsRequestOptions
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DynamicClassifyOptions"/> class.
+        /// </summary>
+        public DynamicClassifyOptions(IList<string> categories)
+        {
+            Categories = (categories is not null) ? categories : new List<string>();
+        }
+
+        /// <summary>
+        /// The type of classification to perform indicating whether the result considers a single category or multiple
+        /// categories. If not set, the service default is used.
+        /// </summary>
+        public ClassificationType? ClassificationType { get; set; }
+
+        /// <summary>
+        /// The categories that a given document can be classified with.
+        /// </summary>
+        public IList<string> Categories { get; internal set; } = new List<string>();
```

Comment:
1. As mentioned in one of my first comments, this should probably be `IReadOnlyList<T>` (or `IReadOnlyCollection<T>`, if more consistent throughout). This jives with our guidelines for constructor parameters (required properties).
2. Do not initialize to a `new List<string>` here. Do so in the constructor since you're taking args anyway.. This is a wasted allocation.
3. If this is for a required parameter, `categories` should never be null. Throw an exception. Use `Argument.AssertNotNullOrEmpty` because it's required, which means it has to be non-empty.